connector/exceptionsconnector/connector_metrics.go

```diff
@@ -132,28 +134,42 @@ func (c *metricsConnector) collectExceptions(ilm pmetric.ScopeMetrics) error {
 	dps := mCalls.Sum().DataPoints()
 	dps.EnsureCapacity(len(c.exceptions))
 	timestamp := pcommon.NewTimestampFromTime(time.Now())
-	for _, val := range c.exceptions {
-		dpCalls := dps.AppendEmpty()
-		dpCalls.SetStartTimestamp(c.startTimestamp)
-		dpCalls.SetTimestamp(timestamp)
-
-		dpCalls.SetIntValue(int64(val.count))
-
-		val.attrs.CopyTo(dpCalls.Attributes())
+	for _, exc := range c.exceptions {
+		dp := dps.AppendEmpty()
+		dp.SetStartTimestamp(c.startTimestamp)
+		dp.SetTimestamp(timestamp)
+		dp.SetIntValue(int64(exc.count))
+		for i := 0; i < exc.exemplars.Len(); i++ {
+			exc.exemplars.At(i).SetTimestamp(timestamp)
+		}
+		exc.exemplars.CopyTo(dp.Exemplars())
```

Comment:
i think we could do `dp.Exemplars().EnsureCapacity(exc.exemplars.Len())` here before calling CopyTo.
I see that CopyTo makes an attempt to meet the needs but it might be better readable code if we explicitly request ensureCapacity